Sources/_StringProcessing/Engine/Processor.swift

```diff
@@ -158,8 +183,8 @@ extension Processor {
   /// - Precondition: `bounds.contains(index) || index == bounds.upperBound`
   /// - Precondition: `index >= currentPosition`
   mutating func resume(at index: Input.Index) {
-    assert(index >= bounds.lowerBound)
-    assert(index <= bounds.upperBound)
+    assert(index >= searchBounds.lowerBound)
+    assert(index <= searchBounds.upperBound)
```

Comment:
The end is a valid index to resume at, since that's where an end of subject anchor would need to match. An instruction that needs to test for a character at the current index would just need to fail (not crash) at that point. We already have the right behavior in place for these.